apps/fishing-map/features/analysis/AnalysisBeforeAfterGraph.tsx

```diff
@@ -0,0 +1,175 @@
+  const dtTick = DateTime.fromMillis(tick).toUTC()
+  const dtStart = DateTime.fromISO(timeComparison.compareStart).toUTC()
+  if (tick !== dtStart.toMillis()) {
+    const diff = dtTick.diff(dtStart, timeComparison.durationType as any).toObject()
+    const diffValue = Math.round(diff[timeComparison.durationType as any])
+    const sign = diffValue > 0 ? '+' : ''
+    return [sign, diffValue].join('')
+  }
+  const date = dtTick.setLocale(i18n.language)
+  let formattedTick = ''
+  switch (timeComparison.durationType) {
+    case 'month':
+      formattedTick = date.toFormat('LLL y')
+      break
+    default:
+      formattedTick = date.toLocaleString(DateTime.DATE_MED_WITH_WEEKDAY)
+      break
+  }
+  return formattedTick
+}
+
+const AnalysisGraphTooltip = (props: any) => {
+  const { payload } = props
+
+  const timeComparison = useSelector(selectAnalysisTimeComparison)
+  const avgLineValue = payload?.find((p) => p.name === 'line')
+  if (!avgLineValue) return null
+
+  const date = DateTime.fromMillis(avgLineValue.payload.date).toUTC().setLocale(i18n.language)
+  return (
+    <div className={styles.tooltipContainer}>
+      <p className={styles.tooltipLabel}>{formatDate(date, timeComparison.durationType)}</p>
+      <span className={styles.tooltipValue}>
+        {formatTooltipValue(avgLineValue.payload.avg as number, avgLineValue.unit as string)}
+      </span>
+    </div>
+  )
+}
+
+const AnalysisBeforeAfterGraph: React.FC<{
+  graphData: ComparisonGraphProps
+  start: string
+  end: string
+}> = (props) => {
+  const { start, end } = props
+  const { timeseries, sublayers } = props.graphData
+  const timeComparison = useSelector(selectAnalysisTimeComparison)
+
+  const dtStart = useMemo(() => {
+    return DateTime.fromISO(timeComparison.compareStart).toUTC()
+  }, [timeComparison.compareStart])
+
+  const range = useMemo(() => {
+    const values = timeseries?.flatMap(({ date, compareDate, min, max }) => {
+      return [
+        {
+          date: DateTime.fromISO(date).toUTC().toMillis(),
+          range: [min[0], max[0]],
+          avg: (max[0] + min[0]) / 2,
+        },
+        {
+          date: DateTime.fromISO(compareDate).toUTC().toMillis(),
+          range: [min[1], max[1]],
+          avg: (max[1] + min[1]) / 2,
+        },
+      ]
+    })
+    values.sort((v1, v2) => v2.date - v1.date)
+
+    return values
+  }, [timeseries])
+
+  const ticks = useMemo(() => {
+    const finalTicks = [dtStart.toMillis()]
+    const startIncrementAt = Math.ceil(timeComparison.duration * 0.35)
+    for (let i = startIncrementAt; i <= timeComparison.duration; i++) {
+      finalTicks.push(
+        dtStart
+          .plus({ [timeComparison.durationType]: i })
+          .toUTC()
+          .toMillis()
+      )
+      finalTicks.push(
+        dtStart
+          .minus({ [timeComparison.durationType]: i })
+          .toUTC()
+          .toMillis()
+      )
+    }
+    finalTicks.sort()
+    return finalTicks
+  }, [timeComparison])
```

Comment:
this is going to be recomputed only when timeComparison changes, but shouldn't we do the same for `dtStart` and `durationType`?